File: @xoutput/api/src/websocket/xbox/XboxInputRequest.ts
import { MessageBase } from '../MessageBase';

export const XboxInputRequestType = 'XboxInput';

export interface XboxInputRequest extends MessageBase<typeof XboxInputRequestType> {
  a: boolean;
  b: boolean;
  x: boolean;
  y: boolean;
  l1: boolean;
  l3: boolean;
  r1: boolean;
  r3: boolean;
  start: boolean;
  back: boolean;
  home: boolean;
  up: boolean;
  down: boolean;
  left: boolean;
  right: boolean;
  lX: number;
  lY: number;
  rX: number;
  rY: number;
  l2: number;
  r2: number;
}

File: XOutput.Api/Websocket/Xbox/XboxInputRequest.cs
namespace XOutput.Websocket.Xbox
{
    public class XboxInputRequest : MessageBase
    {
        public const string MessageType = "XboxInput";

        public XboxInputRequest()
        {
            Type = MessageType;
        }

        public bool? A { get; set; }
        public bool? B { get; set; }
        public bool? X { get; set; }
        public bool? Y { get; set; }
        public bool? L1 { get; set; }
        public bool? L3 { get; set; }
        public bool? R1 { get; set; }
        public bool? R3 { get; set; }
        public bool? Start { get; set; }
        public bool? Back { get; set; }
        public bool? Home { get; set; }
        public bool? Up { get; set; }
        public bool? Down { get; set; }
        public bool? Left { get; set; }
        public bool? Right { get; set; }
        public double? LX { get; set; }
        public double? LY { get; set; }
        public double? RX { get; set; }
        public double? RY { get; set; }
        public double? L2 { get; set; }
        public double? R2 { get; set; }
    }
}
